### PR TITLE
Improve model performance

### PR DESCRIPTION
**Pull request by mohannadbanayosi**
## Context
Experiments intended to improve model training and performance.

## Changes
- new custom combined loss -> `MSELoss` + `StructuralSimilarityIndexMeasure`
- increased initial lr + dynamically decreasing
- weight decay
- early training stopping
- skip connections
- remove `BatchNorm2d`
- deeper model
- `DenoisingAutoencoderMini` for local use and `DenoisingAutoencoder` for training on bigger instances

## Experiments
Increasing the depth of the model didn't bring any improvements, unless accompanied with an increase in the initial resizing. The main issue was that doubling the image dimension meant 4x more memory, so I went on to train on an external GPU powered compute. Below are the results:

Pink on BSDS300 dataset and green on a collection of 409 landscape images taken by me.

### PSNR
train_psnr
val_psnr

### SSIM
train_ssim
val_ssim

### Loss (70% SSIM and 30% MSE)
train_loss
val_loss

### CODE REVIEW

**Comment by mohannadbanayosi** [comment]
Experiments so far yield the following results:
experiments

**Comment by mohannadbanayosi** [comment]
## Results

### Original
image_original

### Noisy image
output_noised_1745477403_best_model

### Denoised
output_denoised_1745477403_best_model
